java/org/contikios/cooja/dialogs/AbstractCompileDialog.java

```diff
@@ -499,24 +499,28 @@ public void actionPerformed(ActionEvent e) {
     final Action compilationSuccessAction = new AbstractAction() {
 			@Override
 			public void actionPerformed(ActionEvent e) {
-        abortMenuItem.setEnabled(false);
-
-        /* Make sure firmware exists */
-        if (!contikiFirmware.exists()) {
-          logger.fatal("Contiki firmware does not exist: " + contikiFirmware.getAbsolutePath());
-          setDialogState(DialogState.AWAITING_COMPILATION);
-          return;
-        }
-        setDialogState(DialogState.COMPILED_FIRMWARE);
+        java.awt.EventQueue.invokeLater(() -> {
```

Comment:
ActionListeners are always called from the AWT dispatch thread. I do not think there is any reason to switch thread again here?

Reply:
`onSuccess` and `onFailure` in CompileContiki.java are called from the thread called "handle compilation results".

Reply:
Then those calls should probably switch to AWT dispatch thread?

Reply:
Fixed.

Reply:
Thanks!